cmd/launcher/svc_windows.go

```diff
@@ -0,0 +1,103 @@
+// +build windows
+
+package main
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/logutil"
+	"github.com/kolide/launcher/pkg/log/eventlog"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+)
+
+// TODO This should be inherited from some setting
+const serviceName = "launcher"
+
+// runWindowsSvc starts launcher as a windows service. This will
+// probably not behave correctly if you start it from the command line.
+func runWindowsSvc(args []string) error {
+	eventLogWriter, err := eventlog.NewWriter(serviceName)
+	if err != nil {
+		return errors.Wrap(err, "create eventlog writer")
+	}
+	defer eventLogWriter.Close()
+
+	logger := eventlog.New(eventLogWriter)
+	level.Debug(logger).Log("msg", "service start requested")
+
+	opts, err := parseOptions(os.Args[2:])
+	if err != nil {
+		level.Info(logger).Log("err", err)
+		os.Exit(1)
+	}
+
+	run := svc.Run
+
+	return run(serviceName, &winSvc{logger: logger, opts: opts})
+}
+
+func runWindowsSvcDebug(args []string) error {
+	logger := logutil.NewCLILogger(true) //interactive
+	level.Debug(logger).Log("msg", "foreground service start requested (debug mode)")
+
+	opts, err := parseOptions(os.Args[2:])
+	if err != nil {
+		level.Info(logger).Log("err", err)
+		os.Exit(1)
+	}
+
+	run := debug.Run
+
+	return run(serviceName, &winSvc{logger: logger, opts: opts})
+}
+
+type winSvc struct {
+	logger log.Logger
+	opts   *options
+}
+
+func (w *winSvc) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+	changes <- svc.Status{State: svc.StartPending}
+	level.Debug(w.logger).Log("msg", "windows service starting")
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		err := runLauncher(ctx, cancel, w.opts, w.logger)
+		if err != nil {
+			level.Info(w.logger).Log("err", err)
+			changes <- svc.Status{State: svc.Stopped, Accepts: cmdsAccepted}
+			os.Exit(1)
+		}
+	}()
+
+	for {
+		select {
+		case c := <-r:
```

Comment:
more of a gripe than anything, but the go's idiomatic single letter var names have yet to resonate with me.

Reply:
Yeah... sometimes I like them, other times I have the same reaction.

But this code was just moved from main_windows.go to svc_windows.go.

Reply:
Yea I saw this after I commented on that. 👍